### PR TITLE
WINDUP-723 Improve TypeReferenceLocation javadoc

### PR DESCRIPTION
**Pull request by OndraZizka**
(14:08:08) mbriskar: ozizka, it will. implements/extends is transitive
(14:08:28) mbriskar: ozizka, even method_call probably is transitive

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/373/
Test PASSed.

File: java-ast/addon/src/main/java/org/jboss/windup/ast/java/data/TypeReferenceLocation.java [hunk 1]
```diff
@@ -83,14 +83,16 @@
...
     IMPLEMENTS_TYPE("Implements type"),
 
     /**
      * A variable initalization expression value. For example, this would resolve to "mypackage.MyConstants.FOO" for the following field declaration:
      *
+     * <pre>
      * int foo = mypackage.MyConstants.FOO;
+     * </pre>
      */
     VARIABLE_INITIALIZER("Variable Initializer");
```
Comment (mbriskar) [inline java-ast/addon/src/main/java/org/jboss/windup/ast/java/data/TypeReferenceLocation.java]: I would also somehow provide examples and this comment: " TYPE is registered as soon as a java class is found, transitively on superclasses. Then Implements/EXTENDS are registered transitively on both, interfaces/classes and also all the method_calls are registered transitively but only transitively on the interfaces (not extensions)"

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1002/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/374/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1003/
Test PASSed.

File: java-ast/addon/src/main/java/org/jboss/windup/ast/java/data/TypeReferenceLocation.java (hunk 1 above)
Comment (OndraZizka) [inline java-ast/addon/src/main/java/org/jboss/windup/ast/java/data/TypeReferenceLocation.java]: +1. Examples of code which is matched by particular location.

**Pull request merged**